src/Arc4u.Standard.OAuth2.Client/Security/Principal/AppPrincipalFactory.cs

```diff
@@ -28,84 +27,72 @@ public class AppPrincipalFactory : IAppPrincipalFactory
         public static readonly string tokenExpirationClaimType = "exp";
         public static readonly string[] ClaimsToExclude = { "exp", "aud", "iss", "iat", "nbf", "acr", "aio", "appidacr", "ipaddr", "scp", "sub", "tid", "uti", "unique_name", "apptype", "appid", "ver", "http://schemas.microsoft.com/ws/2008/06/identity/claims/authenticationinstant", "http://schemas.microsoft.com/identity/claims/scope" };
 
+        private readonly IContainerResolve _container;
+        private readonly INetworkInformation _networkInformation;
+        private readonly ICache _claimsCache;
+        private readonly ICacheKeyGenerator _cacheKeyGenerator;
+        private readonly IApplicationContext _applicationContext;
+        private readonly ILogger<AppPrincipalFactory> _logger;
 
-        private bool copyClaimsFromCache = false;
-        private List<ClaimDto> cachedClaims;
-
-        public AppPrincipalFactory(INetworkInformation networkInformation, ISecureCache claimsCache, ICacheKeyGenerator cacheKeyGenerator, IContainerResolve container, IApplicationContext applicationContext, ILogger<AppPrincipalFactory> logger)
+        public AppPrincipalFactory(IContainerResolve container, INetworkInformation networkInformation, ISecureCache claimsCache, ICacheKeyGenerator cacheKeyGenerator, IApplicationContext applicationContext, ILogger<AppPrincipalFactory> logger)
         {
-            NetworkInformation = networkInformation;
-            ClaimsCache = claimsCache;
-            CacheKeyGenerator = cacheKeyGenerator;
-            Container = container;
-            ApplicationContext = applicationContext;
+            _container = container;
+            _networkInformation = networkInformation;
+            _claimsCache = claimsCache;
+            _cacheKeyGenerator = cacheKeyGenerator;
+            _applicationContext = applicationContext;
             _logger = logger;
         }
 
-        private INetworkInformation NetworkInformation { get; set; }
```

Comment:
As those information are initialized only in the constructor. I propose to change from get; set; to get; init; => this improve the immutability...